cfme/tests/control/test_alerts.py

```diff
@@ -128,54 +123,44 @@ def set_performance_capture_threshold(appliance):
     appliance.set_yaml_config(yaml)
 
 
-@pytest.yield_fixture(scope="module")
-def initialize_provider(provider):
-    # Remove all providers
-    BaseProvider.clear_providers()
-    # Setup the provider we want
-    provider.create(validate_credentials=True, validate_inventory=True, check_existing=True)
-    # Take care of C&U settings
-    if provider.type not in CANDU_PROVIDER_TYPES:
-        yield provider
-    else:
-        try:
-            with server_roles_enabled(
-                    'ems_metrics_coordinator', 'ems_metrics_collector', 'ems_metrics_processor'):
-                candu.enable_all()
-                yield provider
-        finally:
-            candu.disable_all()
-
-
-@pytest.fixture(scope="function")
-def vm_name(request, initialize_provider, full_template):
-    name = "test_alerts_{}".format(fauxfactory.gen_alpha())
-
-    @request.addfinalizer
-    def _cleanup_vm():
-        try:
-            if initialize_provider.mgmt.does_vm_exist(name):
-                initialize_provider.mgmt.delete_vm(name)
-            initialize_provider.refresh_provider_relationships()
-        except Exception as e:
-            logger.exception(e)
-    vm_obj = VM.factory(name, initialize_provider, template_name=full_template["name"])
+@pytest.yield_fixture(scope="function")
+def setup_candu(vm):
+    candu.enable_all()
+    with server_roles_enabled('ems_metrics_coordinator', 'ems_metrics_collector',
+            'ems_metrics_processor'):
+        vm.wait_candu_data_available(timeout=20 * 60)
+        yield
+    candu.disable_all()
+
+
+@pytest.fixture(scope="module")
+def vm_name():
+    return random_vm_name(context="alert")
+
+
+@pytest.yield_fixture(scope="function")
+def vm(vm_name, full_template, provider, setup_one_provider_modscope):
```

Comment:
**REQUIRED** `setup_one_provider_modscope` -> `setup_provider`. Function scope wrt `setup_provider` isn't a problem because if the provider already exists it will simply continue without trying to set it up.